src/common/api.ts

```diff
@@ -24,6 +24,33 @@ import type { PathTypes } from '$common/path'
 import type { Proof } from '$common/pools'
 import type { SQLQueryKeys } from './sql'
 
+/**
+ * Result interface for wallet removal operations
+ * @dev Used to communicate the outcome of a wallet deletion request
+ * @notice This interface ensures consistent response structure across the application
+ */
+export interface WalletRemoveResult {
+  /**
+   * Operation success status
+   * @dev true if wallet was removed, false if operation failed
+   */
+  success: boolean
+
+  /**
+   * List of wallets remaining after removal
+   * @dev Contains metadata for all wallets still in the system
+   * @notice Used to update UI state after wallet removal
+   */
+  remainingWallets: WalletMetadata[]
```

Comment:
i'm not sure that this is the right pattern. check what is done with the rest of the handlers. usually this would just be solved via a diff on the front end.